fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/JcrRdfTools.java

```diff
@@ -405,8 +407,19 @@ private Resource getSkolemizedResource(final IdentifierConverter<Resource, Fedor
         final AnonId id = resource.asResource().getId();
 
         if (!skolemizedBnodeMap.containsKey(id)) {
-            final Node orCreateNode = jcrTools.findOrCreateNode(session, skolemizedId());
+            final String pid = pidMinter.mintPid();
```

Comment:
Why are you not able to move `pidMinter.mintPid()` back into the method `skolemizedId()`, then:
`final String path = skolemizedId();`, and
only call `jcrTools.findOrCreateNode(session, path)` once instead of twice?